### PR TITLE
Launch: Cleanup calypsoifyGutenberg.

### LINKED ISSUE #46839: Focused Launch: set up A/B testing
Currently, the Focused Launch Flow modal is enabled via the temporary `gutenboarding/focused-launch-flow` flag.

The goal is to swap the flag with A/B testing logic

**Acceptance Criteria:**

- [x] After creating a site using `/start` flow, when clicking _Launch_ button in the editor, 50% of users should be redirected to `/start/launch-site` and 50% should start the Focused launch flow with no redirect.
- [x] ~When enabling `gutenboarding/focused-launch-flow` in production we should check where it's being used. For example, in [openCheckout](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/utils.ts#L29-L35) we should rather use `!isGutenboarding` to identify the launch flow we're in.~ Now uses ENABLE_FOCUSED_LAUNCH in backend flag.
- [x] Not going through Calypso iframe would be good in the context of https://github.com/Automattic/wp-calypso/issues/47803 so A/B test code could be run in PHP as described in the recommendation below

**Hint:**

We could follow the pattern already implemented by @Automattic/ganon by adding a filter and get the A/B test variations on the backend:
- Server-side A/B test reference: https://github.com/Automattic/wp-calypso/pull/47896
- Server-side launch injection: D53680-code
- A/B test code: https://github.com/Automattic/wp-calypso/pull/49004 D56270-code
- Experiment: pbxNRc-Ag-p2

### PR DESCRIPTION
**Pull request by yansern**
#### Changes proposed in this Pull Request

Note: Ensure ETK is deployed with changes from https://github.com/Automattic/wp-calypso/pull/49004 before deploying this one.

This PR branches off https://github.com/Automattic/wp-calypso/pull/49004 to cleanup props that are no longer used in `calypsoifyGutenberg`.

The reason for this PR's existence is to keep deploys atomic. https://github.com/Automattic/wp-calypso/pull/49004 is related to ETK and this is related to WBE.

~~TODO: The `currentCalypsoUrl` is the only thing left that keeps Focused Launch tethered to Calypso I think. Need to see how to move that out.~~ This is needed for for "Use A Domain I Own" to identify the correct url on the parent iframe.

These feature flags are also removed:
- gutenboarding/new-launch-mobile
- gutenboarding/feature-picker
- create/persistent-launch-button
- create/focused-launch-flow

#### Testing instructions

* Ensure Step-By-Step Launch & Focused Launch still works.
* Also good to check who is accessing props from `calypsoifyGutenberg` object, most places are still relying on `isGutenboarding` and that is why it is still kept.

Fixes https://github.com/Automattic/wp-calypso/issues/46839 #47803 #50089

### CODE REVIEW

**Comment by yansern** [comment]
Ignore D55672-code, this was generated because the PR has an initial incorrect base branch.

First deploy the WBE, then deploy Calypso.

**Comment by yansern** [comment]
Rebased and this commit https://github.com/Automattic/wp-calypso/pull/49067/commits/3a3d2a6f67aa8cbe026f90d950ab251e710e4f12 was reverted because it was a rebase glitch.

**Comment by ciampo** [comment]
Hey @yansern , do you mind rebasing this branch, since it's become slightly stale (6/7 days ago?)

**Comment by ciampo** [comment]
Waiting until Focused Launch is released

**Review by ciampo — APPROVED** [review]
LGTM

File: apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js
Comment (razvanpapadopol) [inline apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js:800]: Let's not remove this yet since it's still in use: https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts#L71

File: client/gutenberg/editor/calypsoify-iframe.tsx
Comment (razvanpapadopol) [inline client/gutenberg/editor/calypsoify-iframe.tsx:389]: Same as above: https://github.com/Automattic/wp-calypso/pull/49067#discussion_r570304869

File: client/gutenberg/editor/calypsoify-iframe.tsx
Comment (razvanpapadopol) [inline client/gutenberg/editor/calypsoify-iframe.tsx:388]: Shouldn't we also cleanup the feature flags from config files?

Comment (yansern) [reply client/gutenberg/editor/calypsoify-iframe.tsx:388]: Okay so these 3 feature flags are now removed.

File: client/gutenberg/editor/calypsoify-iframe.tsx
Comment (yansern) [reply client/gutenberg/editor/calypsoify-iframe.tsx:389]: Okay so these 3 feature flags are now removed.

**Comment by yansern** [comment]
Removed the feature flags:
- gutenboarding/new-launch-mobile
- gutenboarding/feature-picker
- create/persistent-launch-button
- create/focused-launch-flow

And also codes related to `isFocusedLaunchFlowFeatureFlag` for temp feature flag.

Also rebased the PR.

**Comment by StefanNieuwenhuis** [comment]
> Removed the feature flags:
> 
> * gutenboarding/new-launch-mobile
> * gutenboarding/feature-picker
> * create/persistent-launch-button
> * create/focused-launch-flow
> 
> And also codes related to `isFocusedLaunchFlowFeatureFlag` for temp feature flag.
> 
> Also rebased the PR.

I've created #50089 to document this.

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts
Comment (ciampo) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: I would either:
- add a `return`  clause inside all these ifs blocks, or
- re-write the ifs as a `if (..) { … } else if (…) else { … }` chain, or
- use a `switch` statement instead for all the `launchFlow` checks

Basically, this would be to make it explicit that only one type of launch can happen at the same time.

Bonus question: what does the `dispatch( ‘automattic/launch’ ).enableAnchorFm()` call do? If that line is called, should we still run through the rest of the file, or can we _early `return`_?

File: apps/editing-toolkit/typings/fse/index.d.ts
Comment (ciampo) [inline apps/editing-toolkit/typings/fse/index.d.ts]: Are we missing `anchorFmPodcastId` ? From a quick look, it seems like it could be either `string` or `undefined`

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts
Comment (yansern) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: > add a `return` clause inside all these ifs blocks,

Possible with some wrangling... there is a part at the end which calls `savePost()`. This happens in the background for gutenboardingLaunch and focusedLaunch, whereas, this needs to be happen first before siteLaunch.

We'll kinda end up with repeated `savePost()` but I think the readability will improve.

> use a `switch` statement instead for all the launchFlow checks

Possible too, but I think I'm in favour of separate `if` blocks. Not have to add `break;` is nice.


> Bonus question: what does the dispatch( ‘automattic/launch’ ).enableAnchorFm() call do? If that line is called, should we still run through the rest of the file, or can we early return?

Anchor-flavoured sites skip some steps in the gutenboarding launch flow (and they are exclusively used the gutenboarding launch flow). `enableAnchorFm` toggles a flag in the data store that changes the step sequence, then the step-by-step launch opens.

File: apps/editing-toolkit/typings/fse/index.d.ts
Comment (yansern) [reply apps/editing-toolkit/typings/fse/index.d.ts]: Thanks! Will add it in.

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts
Comment (ciampo) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: I see, thank you for the explanation

> there is a part at the end which calls savePost(). This happens in the background for gutenboardingLaunch and focusedLaunch, whereas, this needs to be happen first before siteLaunch.

I missed that part! In that case, we can probably leave it as is — I'm not sure that any refactor would improve the code clarity much

Comment (yansern) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: I'm going to leave the `switch` version here for reference.
I'm not sure why there was a `setTimeout` on the `savePost()` for gutenboarding launch & focused launch.
It's either because we don't want user to see the post is saving or we don't want 2 things to happen at once (slow down the launch popup?).
This switch version reuses one `savePost()` for all without the `setTimeout`.

```js
const saveTask = dispatch( 'core/editor' ).savePost();

switch ( launchFlow ) {
	case GUTENBOARDING_LAUNCH_FLOW:
		dispatch( 'automattic/launch' ).openSidebar();
		break;

	case FOCUSED_LAUNCH_FLOW:
		dispatch( 'automattic/launch' ).openFocusedLaunch();
		break;

	case SITE_LAUNCH_FLOW:
		// Save post first before redirecting to launch url
		( async () => {
			await saveTask;
			window.top.location.href = launchUrl;
		} )();
		break;
}
```

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts
Comment (yansern) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts:51]: Though unlikely, this *might* be one reason users assigned to focused launch still end up with site launch.

Comment (yansern) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: Explanation for this change: Though unlikely, this **might** be the reason users assigned to focused launch still end up with site launch.

Comment (ciampo) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: Why don't we make it a `<button>` instead? It would make it more semantically correct and avoid the same issues (plus, this button is already injected via JS)

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts
Comment (ciampo) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: I think that the delay was there to:
- prioritise any API call that the launch modal would make
- avoid showing the "post saved" toast in the bottom left of the screen (by adding a delay, the toast would appear while the launch modal is already visible, and therefore the user would not see it)

This is one possible way of using a `switch` statement while keeping the `setTimeout` call

```typescript
const { savePost } = dispatch( 'core/editor' );
const delayedSavePost = () => setTimeout( () => savePost(), 1000 );

switch ( launchFlow ) {
	case GUTENBOARDING_LAUNCH_FLOW:
		// Save post in the background while step-by-step flow opens
		dispatch( 'automattic/launch' ).openSidebar();
		delayedSavePost();
		break;
	case FOCUSED_LAUNCH_FLOW:
		// Save post in the background while focused launch flow opens
		dispatch( 'automattic/launch' ).openFocusedLaunch();
		delayedSavePost();
		break;
	case SITE_LAUNCH_FLOW:
        // Save post first before redirecting to launch url
		( async () => {
			await savePost();
			window.top.location.href = launchUrl;
		} )();
		break;
}
```

Comment (ciampo) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: Actually, I managed to track down the PR that introduced the 1s delay #44909 — @razvanpapadopol , is there a reason why we introduced that arbitrary delay?

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts
Comment (yansern) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: Good point. Changing it to a button.

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts
Comment (yansern) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: Thanks for the `switch` version which seems like a good middle ground for now. :) Delaying save post does have its benefit regardless of the reason, and this block of code is readable, much readable than the existing one where it's executed at the bottom. I'm committing this in.

Comment (razvanpapadopol) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: I don't remember exactly why it's 1s delay. Maybe something related to how long it would take for the launch modal to open? Eg: with a slide-in animation

**Comment by ciampo** [comment]
Rebased on top of trunk

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts
Comment (ciampo) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: Alright — let's keep the 1s delay in, for now

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts
Comment (ciampo) [reply apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/launch-button/index.ts]: Awesome, thank you!

File: apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js
Comment (ciampo) [reply apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js:800]: Resolving the conversation, since the use of the `isFocusedLaunchFlow` is being removed with this same PR now

**Review by ciampo — APPROVED** [review]
Overall, the code looks much cleaner! Awesome job @yansern 


I gave it another round of testing after recent changes:

- [x] Step by Step launch modal opens correctly
- [x] Focused Launch modal opens correctly when A/B testing assigned to "treatment"
- [x] Focused Launch modal is not affected by the `create/focused-launch-flow` feature flag
- [x] Control Launch flow happens when Focused Launch A/B testing assigned to "control"


I guess the order of operations is going to be:

1. ~Deploy WBE~ Actually not necessary, as the changes to be deploy are just deletions of deprecated properties 
2. Merge Calypso
3. Release ETK

**Comment by ciampo** [comment]
Let's wait until #48452 is merged, rebase on top of trunk, and see if there's any changes needed (e.g. related to `getCurrentLaunchFlowUrl`)

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/utils.ts
Comment (yansern) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/utils.ts]: Swithed over to `??` version after rebase. cc @ciampo

File: apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-focused-launch.tsx
Comment (ciampo) [inline apps/editing-toolkit/editing-toolkit-plugin/editor-site-launch/src/attach-focused-launch.tsx:47]: The `inIframe()` check was redundant, since `getCurrentLaunchFlowUrl()` already uses `window.location.href` when in Wp Admin

**Pull request merged**
